internal/security/kdf/methods.go

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package kdf
+
+import (
+	"crypto/rand"
+	"errors"
+	"hash"
+	"io"
+	"os"
+	"path"
+
+	"golang.org/x/crypto/hkdf"
+
+	"github.com/edgexfoundry/go-mod-secrets/pkg/token/fileioperformer"
+)
+
+const (
+	saltFile   string = "kdf-salt.dat"
+	saltLength        = 32
+)
+
+// file implements a unit testing hook for file IO
+type file interface {
+	io.Reader
+	io.Closer
+	io.WriterAt
```

Comment:
if you don't need specifically WriteAt (which IMHO you don't), then you could get rid of this whole interface and just use `io.ReadWriteCloser` instead :tada:

Reply:
Fixed.